tests/test_requests.py

```diff
@@ -234,12 +234,52 @@ def test_http_301_changes_post_to_get(self, httpbin):
 
     def test_http_301_doesnt_change_head_to_get(self, httpbin):
         r = requests.head(httpbin('status', '301'), allow_redirects=True)
-        print(r.content)
         assert r.status_code == 200
         assert r.request.method == 'HEAD'
         assert r.history[0].status_code == 301
         assert r.history[0].is_redirect
 
+    def test_http_301_doesnt_change_non_post_to_get(self, httpbin):
+        r = requests.patch(httpbin('redirect-to'),
+                           data='test body',
+                           params={'url': 'patch', 'status_code': '301'})
+        assert r.status_code == 200
+        assert r.request.method == 'PATCH'
+        assert r.history[0].status_code == 301
+        assert r.history[0].is_redirect
+        assert r.request.body == 'test body'
+        assert r.json()['data'] == 'test body'
```

Comment:
You're right, this one was written before `test_http_301_for_redirectable_methods` was, following the styling above. We can actually collapse the three tests above into the one below, but I to ensure we were always testing for those specific cases. The thought being if `test_http_301_for_redirectable_methods` is later modified it would be nice for those to still fail, but I understand if you'd like to see them squashed together.

As for the status code, I left that out of the test below because /head doesn't exist in httpbin (because it's non-sensical). The test logic becomes more complex trying to code around the HEAD case and the status code isn't needed confirm we got what we wanted from the test, only that httpbin is functioning properly.